apps/extension/src/core/views/ActivityItem.tsx

```diff
@@ -1,13 +1,61 @@
+import {
+  MediaMimeType,
+  isMediaOutput,
+  isPromptInput,
+  mimeTypeToExtension,
+  isMediaMimeType
+} from "window.ai"
+
 import { Logo } from "~core/components/pure/Logo"
 import { Text } from "~core/components/pure/Text"
 import { originManager } from "~core/managers/origin"
 import type { Transaction } from "~core/managers/transaction"
 import { transactionManager } from "~core/managers/transaction"
 import { formatDate } from "~core/utils/utils"
 
+function createMediaDownloadLinks(transaction: Transaction) {
+  if (!transaction.input || !transaction.outputs) {
+    return null
+  }
+  const input = transaction.input
+  // fallback to .txt containing contents
+  let mimeType = MediaMimeType.TXT
```

Comment:
if every output is a media output, then why initialize or allow mimeType to be a TXT mime type?

Reply:
make mimeTypeToExtension into a function, handle undefined case easily

Reply:
changed to not use TXT types